lib/ui/text/asset_manager_font_provider.cc

```diff
@@ -0,0 +1,119 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/lib/ui/text/asset_manager_font_provider.h"
+
+#include "lib/fxl/logging.h"
+#include "third_party/skia/include/core/SkData.h"
+#include "third_party/skia/include/core/SkStream.h"
+#include "third_party/skia/include/core/SkString.h"
+#include "third_party/skia/include/core/SkTypeface.h"
+
+namespace blink {
+
+namespace {
+
+void VectorReleaseProc(const void* ptr, void* context) {
+  delete reinterpret_cast<std::vector<uint8_t>*>(context);
+}
+
+}  // anonymous namespace
+
+AssetManagerFontProvider::AssetManagerFontProvider(
+    fml::RefPtr<blink::AssetManager> asset_manager)
+    : asset_manager_(asset_manager) {}
+
+AssetManagerFontProvider::~AssetManagerFontProvider() = default;
+
+size_t AssetManagerFontProvider::GetFamilyCount() {
+  return family_names_.size();
+}
+
+std::string AssetManagerFontProvider::GetFamilyName(int index) {
+  return family_names_[index];
+}
+
+SkFontStyleSet* AssetManagerFontProvider::MatchFamily(
```

Comment:
Nit: I know we don't do this consistently, but can we start putting the superclass this adds an override for in a comment above similar to how we decorate the headers? IMO, it makes things much clearer.

Reply:
done